grammars/c.gbnf

```diff
@@ -0,0 +1,42 @@
+root ::= (declaration)*
+
+declaration ::= dataType identifier "(" parameter? ")" "{" statement* "}"
+
+dataType  ::= "int" ws | "float" ws | "char" ws
+identifier ::= [a-zA-Z_] [a-zA-Z_0-9]*
```

Comment:
This matches keywords, which, with the function call syntax, allows the model to produce things like `while (n1/n2);` (`llama-2-13b` did this quite a bit in my testing, although it's possible some of that was it trying to use syntaxes missing from this grammar). I don't have a great solution for this, other than over-constraining identifiers (e.g. forcing them to start with "n_"), or a complicated rule to precisely match character sequences other than "if", "for" etc. I came up with this hack for testing, which is of course overly restrictive:
```
identifier ::= [a-eghj-qs-vxyz] [a-zA-Z_0-9]*
```
We may need a grammar feature for this.